builder/job.go

```diff
@@ -142,6 +143,16 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 
 	sf := streamformatter.NewStreamFormatter(buildConfig.JSONFormat)
 
+	var defaultArchiver *archive.Archiver
+	uidMaps, gidMaps := d.GetUidGidMaps()
+	if uidMaps != nil || gidMaps != nil {
```

Comment:
It's a good question--right now I have some "laziness" in the code which means that when the daemon has been given a `--root=` setting, there will **always** be valid maps for both uid and gid.  Meaning in today's codebase it would be enough to only check one, and if one is non-nil, you are guaranteed the other is non-nil.   I was tempted to have a Daemon boolean config value for user namespaces as it would be simpler to check "*is user namespace turned on?*" throughout the code rather than an *OR* or *AND* of the two possible maps every time.

If you or any other reviewer has strong feelings about the model for checking on/off, let's discuss here.  Put another way, if we are going to create a process with a user namespace, I believe it will require both a user and group ID map, so I don't think there will ever be a model to support only one being non-nil.